packages/core/src/components/StepList/StepList.jsx

```diff
@@ -5,7 +5,11 @@ import Step from './Step';
 export const StepList = ({ steps, ...props }) => (
   <ol className="ds-c-step-list ds-u-margin-top--4">
     {steps.map((step, i) => (
-      <Step step={step} key={step.id || i} {...props} />
+      <Step
```

Comment:
I defined 2 ways to use `component` in `<StepList>` and `VerticalNav`. It can either be set on a granular level within each specific `step` or `item` object, or it can be set on higher levels of `step` or `item`. 

I.e., you can set `component` once on the root `VerticalNav` component in order to make every single nav item to use the passed in component, OR you can set it on a single child if you only want one nav item to have a custom component. 

I'm thinking for the react-router example, it wont be necessary to have such granular control. Should I just keep it simple and have 1 `component` prop on the root element?